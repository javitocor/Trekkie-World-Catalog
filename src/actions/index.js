import callAPI from '../helpers/API';

export const GET_EPISODES = 'GET EPISODES';
export const GET_SEASONS = 'GET SEASONS';
export const CHANGE_FILTER = 'CHANGE FILTER';

export const getEpisodes = show => async dispatch => {
  callAPI(show).then(data => {
    dispatch({
      type: GET_EPISODES,
      data,
    });
  });
};

export const getSeasons = show => dispatch => {
  callAPI(show, 0).then(data => {
    dispatch({
      type: GET_SEASONS,
      data,
    });
  });
};

export const changeFilter = filter => ({
  type: CHANGE_FILTER,
  filter,
});
