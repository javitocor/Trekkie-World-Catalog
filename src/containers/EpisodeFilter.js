import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import episodeFilter from '../style/EpisodeFilter.module.css';

function EpisodeFilter(props) {
  const { handleFilterChange, seasons } = props;
  const seasonsLength = seasons.length;
  const NumSeasons = [...Array(seasonsLength)].map((_, i) => i + 1);
  const dropMenu = NumSeasons.map(
    number => (
      <option key={number} value={number}>
        Season
        {' '}
        {number}
      </option>
    ),
  );
  return (
    <div className={episodeFilter.form}>
      <label htmlFor="filter">Filter by Season</label>
      <select className="" name="filter" id="filter" onChange={event => handleFilterChange(event)}>
        <option value="All">All</option>
        {dropMenu}
      </select>
    </div>
  );
}

EpisodeFilter.propTypes = {
  handleFilterChange: PropTypes.func.isRequired,
  seasons: PropTypes.instanceOf(Object).isRequired,
};

const mapStateToProps = state => ({
  seasons: state.seasons,
});

export default connect(mapStateToProps)(EpisodeFilter);
