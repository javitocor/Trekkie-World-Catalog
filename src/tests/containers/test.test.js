/* eslint-disable no-unused-vars */
import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';

import EpisodeFilter from '../../containers/EpisodeFilter';
import { changeFilter } from '../../actions/index';

const mockStore = configureStore([]);

describe('My Connected React-Redux Component', () => {
  let store;
  let component;

  beforeEach(() => {
    store = mockStore({
      seasons: [{
        name: 'TOS',
      }],
    });

    store.dispatch = jest.fn();

    component = renderer.create(
      <Provider store={store}>
        <EpisodeFilter />
      </Provider>,
    );
  });
  it('should render with given state from Redux store', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
