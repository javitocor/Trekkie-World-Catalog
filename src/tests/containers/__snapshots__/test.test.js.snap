// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Connected React-Redux Component should render with given state from Redux store 1`] = `
<div
  className="form"
>
  <label
    htmlFor="filter"
  >
    Filter by Season
  </label>
  <select
    className=""
    data-testid="select"
    id="filter"
    name="filter"
    onChange={[Function]}
  >
    <option
      data-testid="select-option"
      value="All"
    >
      All
    </option>
    <option
      data-testid="select-option"
      value={1}
    >
      Season
       
      1
    </option>
  </select>
</div>
`;
